api/client/build.go

```diff
@@ -265,6 +268,14 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	headers.Add("X-Registry-Config", base64.URLEncoding.EncodeToString(buf))
 	headers.Set("Content-Type", "application/tar")
 
+	// collect all the build-time environment variables for the container
+	buildArgs := runconfig.ConvertKVStringsToMap(flBuildArg.GetAll())
+	buf, err = json.Marshal(buildArgs)
+	if err != nil {
+		return err
+	}
+	headers.Add("X-Docker-BuildArgs", base64.URLEncoding.EncodeToString(buf))
```

Comment:
See how ulimits are passed in here: https://github.com/docker/docker/blob/master/api/client/build.go#L253-L258
Are json encoded into the URL instead of json+base64 in the header.